py/desispec/validredshifts.py

```diff
@@ -0,0 +1,152 @@
+# Example usage:
+# redrock_path = '/global/cfs/cdirs/desi/spectro/redux/guadalupe/tiles/cumulative/1392/20210517/redrock-5-1392-thru20210517.fits'
+# cat = validate(redrock_path, return_target_columns=True)
+
+import os, warnings
+import numpy as np
+from astropy.table import Table, hstack, join
+import fitsio
+
+
+def get_good_fiberstatus(cat, isqso=False):
+    if not isqso:
+        good_fiberstatus = cat['COADD_FIBERSTATUS']==0
+    else:
+        good_fiberstatus = (cat['COADD_FIBERSTATUS']==0) | (cat['COADD_FIBERSTATUS']==8388608) | (cat['COADD_FIBERSTATUS']==16777216)
+    good_fiberstatus &= cat['ZWARN'] & 2**9 == 0  # NO DATA flag
+    return good_fiberstatus
+
+
+def validate(redrock_path, fiberstatus_cut=True, return_target_columns=False, extra_columns=None, emline_path=None, ignore_emline=False):
```

Comment:
For flexibility, we usually try to split algoritms from I/O.  Please split this into two functions:
one function that takes the redrock_path + other options, reads the files it needs, and then calls a second function that takes tables as inputs and does the filtering without doing any I/O.